apps/server/src/modules/tldraw/repo/tldraw-board.repo.ts

```diff
@@ -31,24 +41,26 @@ export class TldrawBoardRepo {
 		}
 	}
 
-	public async updateDocument(docName: string, ydoc: WsSharedDocDo): Promise<void> {
+	public async loadDocument(docName: string, ydoc: WsSharedDocDo): Promise<void> {
 		const persistedYdoc = await this.getYDocFromMdb(docName);
 		const persistedStateVector = encodeStateVector(persistedYdoc);
 		const diff = encodeStateAsUpdate(ydoc, persistedStateVector);
 		await this.updateStoredDocWithDiff(docName, diff);
-
 		applyUpdate(ydoc, encodeStateAsUpdate(persistedYdoc));
 
-		ydoc.on('update', (update: Uint8Array) => this.mdb.storeUpdateTransactional(docName, update));
-
 		persistedYdoc.destroy();
 	}
 
-	public async flushDocument(docName: string): Promise<void> {
-		await this.mdb.flushDocumentTransactional(docName);
+	public async compressDocument(docName: string): Promise<void> {
+		await this.mdb.compressDocumentTransactional(docName);
 	}
 
 	public async storeUpdate(docName: string, update: Uint8Array): Promise<void> {
 		await this.mdb.storeUpdateTransactional(docName, update);
+
+		const currentClock = await this.mdb.getCurrentUpdateClock(docName);
```

Comment:
I am not really sure if i understand what is does. How often it is called now? By each change/update?

Reply:
yes, basically get the latest document from collection and read its clock

Reply:
removed as I actually already get current clock from storeUpdateTransactional